pkg/kubectl/cmd/apply_test.go

```diff
@@ -550,6 +551,56 @@ func TestApplyRetry(t *testing.T) {
 	}
 }
 
+func TestApplyNoChange(t *testing.T) {
+	initTestErrorHandler(t)
+	nameRC, currentRC := readAndAnnotateReplicationController(t, filenameRC)
+	pathRC := "/namespaces/test/replicationcontrollers/" + nameRC
+
+	patchCount := 0
+	f, tf, _, _ := cmdtesting.NewAPIFactory()
+	tf.Printer = &testPrinter{}
+	tf.UnstructuredClient = &fake.RESTClient{
+		APIRegistry:          api.Registry,
+		NegotiatedSerializer: unstructuredSerializer,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			switch p, m := req.URL.Path, req.Method; {
+			case p == pathRC && m == "GET":
+				file, err := os.Open(filenameRCAnnotationJSON)
```

Comment:
Isn't this identical to `currentRC`? Can you just reuse that?

Reply:
No, currentRC have annotation different from the rc. So it  patch with not empty request

Reply:
I see. There is too much setup to this test, so I'm not convinced that it's actually verifying the functionality.  Here is my proposal:

1. Make the first apply call
  a. Initial get should return a 404 (see `TestApplyNonExistObject`)
  b. Expect a POST request, this is the create corresponding to the initial apply, since the obect wasn't there. Save the request body.
2. Make a second apply call
  a. This time, the GET handler returns the body captured in 1.b
  b. That's it - no need to count the patches

Benefits to this approach include:
- no need to add more manifests to the test
- It's testing against the expected scenario

Reply:
Yes, your suggestion may be easier. It is really hard to construct an empty patch